### ISSUE TITLE
dynamic playlists don't allow mixing AND and OR arguments

### ISSUE THREAD

**Issue by Clementine-Issue-Importer** [clementine-player/Clementine#1026]
_From [gobnuts](https://code.google.com/u/104032811542611505172/) on December 05, 2010 21:53:21_

<rip from Issue 70 >
I cannot create a new smart playlist in which I have the following settings (for example):

One of the following is true:
genre contains: Genre A 
OR
genre contains: Genre B

AND
the following is true:
Rating is greater than: 3 

To sum it up: I can use the three arguments connected by AND or by OR, but not mix the arguments. 
I guess a dynamic playlist with the parameters 
"genre a OR b OR c" AND "rating>2 OR rating<5"
is rather common. 

</rip >

This could be done by basically duplicating what's there already. 

One set is for AND, the other is for OR arguments, both are (automatically connected by AND

basically, the way it was in amarok
have to sections, one were all  terms have to match (AND), and another were one has to match (OR)
Example Image: http://www.freeimagehosting.net/image.php?71f5c45084.png (the first checkbox has the OR arguments, the second one the AND arguments

_Original issue: http://code.google.com/p/clementine-player/issues/detail?id=1026_

**Comment by coconutlulz** [comment]
Yes, I would like to do this.

It seems as though it would be simplest to accept raw SQL, which can then be sanitised and executed following the 'WHERE' clause of the query. A more human-readable version could be implemented, but is it worth the effort?

**Comment by stanislavkazmin** [comment]
Yeah, this is what I missing in Clementine for now. Would be nice, if this would be implemented in any (most simple) way.
